test/models/user_test.rb

```diff
@@ -26,6 +26,12 @@ def setup
     refute User.new(email: test_email).valid?
   end
 
+  test 'email must be saved downcased' do
+    new_email = 'NEW_EMAIL@exaMple.cOm'
+    assert create(:user, email: new_email.downcase)
+    refute User.new(email: new_email).valid?
+  end
+
```

Comment:
@sabrown84 - You're on the right track.   Here's some pseudo code for how you'd want the test to go:

```
test 'email is downcased on create' do
  address = 'NEW_EMAIL@exaMple.cOm'
  user = create (not build) a user with that address
  reload the user (may or may not need this)
  assert that the user's email == downcased version of address
end
```

That test's the case when a `User` is being created.  A good second test would be when a `User` is being updated.

```
test 'email is downcased on update' do
  user = create a user with that address
  address = 'NEW_EMAIL@exaMple.cOm'
  user's address = address
  save the user
  assert that the user's email == downcased version of address
end
```

[assert_equal](https://apidock.com/ruby/v1_9_3_392/Test/Unit/Assertions/assert_equal) is a good assertion method to use for these.